src/txmempool.cpp

```diff
@@ -187,14 +170,20 @@ bool CTxMemPool::CalculateMemPoolAncestors(const CTxMemPoolEntry &entry, setEntr
         staged_ancestors.erase(stage);
         totalSizeWithAncestors += stageit->GetTxSize();
 
-        if (stageit->GetSizeWithDescendants() + entry.GetTxSize() > limitDescendantSize) {
-            errString = strprintf("exceeds descendant size limit for tx %s [limit: %u]", stageit->GetTx().GetHash().ToString(), limitDescendantSize);
+        if (stageit->GetSizeWithDescendants() + entry_size > limitDescendantSize) {
+            errString = strprintf("%sexceeds descendant size limit for tx %s [limit: %u]",
+                                  entry_count > 1 ? "possibly " : "",
```

Comment:
These conditional log changes make the function less generic (it now knows something semantic about the caller). If you really want this "possibly " prefix, you could prepend it in the calling function:

```diff
     // When multiple transactions are passed in, the ancestors and descendants of all transactions
     // considered together must be within limits even if they are not interdependent. This may be
     // stricter than the limits for each individual transaction.
-    return calculateAncestorsAndCheckLimits(total_size, package.size(),
-                                            setAncestors, staged_ancestors,
-                                            limitAncestorCount, limitAncestorSize,
-                                            limitDescendantCount, limitDescendantSize, errString);
+    auto ret = calculateAncestorsAndCheckLimits(total_size, package.size(),
+                                                setAncestors, staged_ancestors,
+                                                limitAncestorCount, limitAncestorSize,
+                                                limitDescendantCount, limitDescendantSize, errString);
+
+    errString.insert(0, "possibly ");
+    return ret;
 }
```

Although I don't think you really need it. If someone is submitting packages through the RPC, then presumably they're a sophisticated user and they'll be able to understand that a heuristic is being used to calculate ancestor/descendant limits.

You could also update the only place that this string is used:

```diff
--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -1094,7 +1094,7 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptMultipleTransactions(const std::
             // be implementation-dependent, and it's likely to be multiple transactions because we
             // evaluated all of them together. Return the same failure for all transactions.
             for (auto& ws : workspaces) {
-                ws.m_state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "exceeds-ancestor-descendant-limits", err_string);
+                ws.m_state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "exceeds-ancestor-descendant-limits-heuristic", err_string);
                 results.emplace(ws.m_ptx->GetWitnessHash(), MempoolAcceptResult::Failure(ws.m_state));
             }
             package_state.Invalid(PackageValidationResult::PCKG_POLICY, "package-mempool-limits");
```

Reply:
Good point, added the "possibly " prefix outside